src/Compiler/pars.fsy

```diff
@@ -5718,6 +5722,32 @@ topAppType:
   | appType
      { $1, SynArgInfo([], false, None) }
 
+/* Grammar rule meant for recovery scenarios */
+/* For example in unionCaseReprElement where function type is not allowed */
+invalidUseOfAppTypeFunction:
```

Comment:
It would probably be nice to also use it in some other cases where `appType` is currently used and where a user may mistakenly type a function type.